systemdspawner/systemdspawner.py

```diff
@@ -147,25 +149,55 @@ class SystemdSpawner(Spawner):
         """
     ).tag(config=True)
 
+    unit_trailing_character = Unicode(
+        '-',
+        help="""
+        When using a unit that ends in a SERVERNAME strip this character from the end of the parsed string. 
+        """
+    ).tag(config=True)
+
+    escape_char = Unicode(
+        ':',
+        help="""
+        The character to change invalid safe characters to.
+        """
+    ).tag(config=True)
+
+    # Characters that are safe for systemd units.
+    safe_chars = set(string.ascii_lowercase + string.digits + string.ascii_uppercase + ':-_.\\')
```

Comment:
The escape character cannot be in the list of safe chars. Not sure if this causes an error or a warning, but it's best to remove it.